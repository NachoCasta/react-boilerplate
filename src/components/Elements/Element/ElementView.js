import React from "react";
import {
    CardContainer,
    Flex,
    Box,
    EditableTextBlock,
    Divider,
    Button
} from "components/UI";

const Element = ({
    edit,
    title,
    onTitleChange,
    description,
    onDescriptionChange,
    content,
    onContentChange,
    onEdit,
    onCancel,
    onSave,
    onDelete
}) => (
    <CardContainer
        title={
            <EditableTextBlock
                edit={edit}
                value={title}
                onChange={onTitleChange}
                autosize={ { maxRows: 1 } }
            />
        }
        heading
    >
        <Flex justifyContent="flex-end">
            {edit ? (
                <React.Fragment>
                    <Box pr={1}>
                        <Button type="primary" icon="save" onClick={onSave}>
                            Save
                        </Button>
                    </Box>
                    <Box>
                        <Button type="danger" icon="close" onClick={onCancel}>
                            Cancel
                        </Button>
                    </Box>
                </React.Fragment>
            ) : (
                <React.Fragment>
                    <Box pr={1}>
                        <Button type="primary" icon="edit" onClick={onEdit}>
                            Edit
                        </Button>
                    </Box>
                    <Box>
                        <Button type="danger" icon="delete" onClick={onDelete}>
                            Delete
                        </Button>
                    </Box>
                </React.Fragment>
            )}
        </Flex>
        <Flex>
            <Box width={1} p={1}>
                <EditableTextBlock
                    title="Description"
                    edit={edit}
                    value={description}
                    onChange={onDescriptionChange}
                />
            </Box>
            <Divider />
            <Box width={1} p={1}>
                <EditableTextBlock
                    title="Content"
                    edit={edit}
                    value={content}
                    onChange={onContentChange}
                />
            </Box>
        </Flex>
    </CardContainer>
);

export default Element;
