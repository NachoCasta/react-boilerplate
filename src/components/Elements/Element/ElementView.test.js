import React from "react";
import { shallow } from "enzyme";

import ElementView from "./ElementView";

describe("ElementView", () => {
    let component;
    beforeEach(() => {
        component = shallow(<ElementView />);
    });

    it("renders", () => {
        shallow(<ElementView />);
    });

    it("matches snapshot", () => {
        expect(component).toMatchSnapshot();
    });
});
