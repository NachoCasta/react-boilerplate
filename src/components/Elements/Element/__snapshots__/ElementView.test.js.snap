// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ElementView matches snapshot 1`] = `ShallowWrapper {}`;
